File: app/src/main/java/com/vocaby/app/utils/exception/IllegalFileException.java
package com.vocaby.app.utils.exception;

public class IllegalFileException extends Exception {
    public static final int INVALID_FORMAT = 0;
    public static final int INVALID_FILE = 1;
    private final int code;

    public IllegalFileException(int code) {
        super();
        this.code = code;
    }

    public IllegalFileException(String message, int code) {
        super(message);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: app/src/main/java/com/vocaby/app/exceptions/IllegalFileException.kt
package com.vocaby.app.exceptions

class IllegalFileException : Exception {
    val code: Int

    constructor(code: Int) : super() {
        this.code = code
    }

    constructor(message: String?, code: Int) : super(message) {
        this.code = code
    }

    companion object {
        const val INVALID_FORMAT = 0
        const val INVALID_FILE = 1
    }
}